src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,22 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super(popupSelector);
+    this._popup = document.querySelector(popupSelector);
```

Comment:
`    this._popup = document.querySelector(popupSelector);
`

Please, delete it

**No** need to search the **modal** again in the inheritors of class `Popup` because the parent has already found it in its constructor as `this._popupElement`. It’s available in any child.

Reply:
Thank you! Fixed on main branch.